op-acceptance-tests/tests/interop/interop_txplan_test.go

```diff
@@ -406,8 +410,126 @@ func executeMessageInvalidAttributes(
+					require.NoError(t, err)
+					mu.Lock()
+					chainWalletsOpts = append(chainWalletsOpts, chainWalletOpts)
+					mu.Unlock()
+				}(chainIdx)
+			}
+			wg.Wait()
+		}
+
+		// runPubSubPair spawns publisher goroutine, paired with subscriber goroutine
+		runPubSubPair := func(pubOpt, subOpt txplan.Option, eventLoggerAddress common.Address, localRng *rand.Rand) {
+			var wg sync.WaitGroup
+			wg.Add(2)
+
+			ch := make(chan *txintent.IntentTx[*txintent.MultiTrigger, *txintent.InteropOutput])
+
+			// publisher initiates txCnt transactions that includes multiple random messages
+			publisherRng := rand.New(rand.NewSource(localRng.Int63()))
+			go func(ch chan<- *txintent.IntentTx[*txintent.MultiTrigger, *txintent.InteropOutput], rng *rand.Rand) {
+				defer wg.Done()
+				for range txCnt {
+					tx, receipt, err := InitiateRandomMessages(ctx, pubOpt, rng, eventLoggerAddress)
+					require.NoError(t, err)
+					logger.Info("initiate messages included", "chainID", tx.PlannedTx.ChainID.Value().String(), "blockNumber", receipt.BlockNumber, "block", receipt.BlockHash)
+					ch <- tx
+					time.Sleep(time.Duration(rng.Intn(250)) * time.Millisecond)
+				}
+				close(ch)
+			}(ch, publisherRng)
+
+			subscriberRng := rand.New(rand.NewSource(localRng.Int63()))
+			// subscriber validates every messages that was initiated by the publisher
+			go func(ch <-chan *txintent.IntentTx[*txintent.MultiTrigger, *txintent.InteropOutput], rng *rand.Rand) {
+				defer wg.Done()
+				for dependsOn := range ch {
+					tx, receipt, err := ValidateEveryMessages(ctx, subOpt, dependsOn)
+					require.NoError(t, err)
+					logger.Info("validate messages included", "blockNumber", receipt.BlockNumber, "block", receipt.BlockHash)
+					logger.Info("message dependency",
+						"sourceChainID", dependsOn.PlannedTx.ChainID.Value().String(),
+						"destChainID", tx.PlannedTx.ChainID.Value().String(),
+						"sourceBlockNum", dependsOn.PlannedTx.IncludedBlock.Value().Number,
+						"destBlockNum", receipt.BlockNumber)
+					time.Sleep(time.Duration(rng.Intn(250)) * time.Millisecond)
+				}
+			}(ch, subscriberRng)
+
+			wg.Wait()
+		}
+
+		runPubSubPairWrapper := func(sourceIdx, destIdx, pairIdx int, wg *sync.WaitGroup, localRng *rand.Rand) {
+			defer wg.Done()
+			runPubSubPair(chainWalletsOpts[sourceIdx][pairIdx], chainWalletsOpts[destIdx][pairIdx], eventLoggerAddresses[sourceIdx], localRng)
+		}
+
+		var wg sync.WaitGroup
+		wg.Add(pubSubPairCnt)
+		for pairIdx := range pubSubPairCnt {
+			// randomize source and destination L2 chains
+			perm := rng.Perm(l2ChainNums)
+			sourceIdx, destIdx := perm[0], perm[1]
+			// localRng is needed per pubsub pair because rng cannot be shared without mutex
+			localRng := rand.New(rand.NewSource(rng.Int63()))
+			go runPubSubPairWrapper(sourceIdx, destIdx, pairIdx, &wg, localRng)
+		}
+		wg.Wait()
+	}
+}
+
 func TestInteropTxTest(t *testing.T) {
-	l2ChainNums := 2
+	l2ChainNums := 3
```

Comment:
See comment in kurtosis file, I think 2 chains is better for now

Reply:
Fixed at https://github.com/ethereum-optimism/optimism/pull/15049/commits/524a9971abb0761eeff357aee8c221066f289d29